notebooks/problem-sets/quantum_enigma_deutsch.ipynb

```diff
@@ -0,0 +1,382 @@
+    "\n",
+    "qc = QuantumCircuit(2)\n",
+    "\n",
+    "qc.h(0)\n",
+    "qc.x(1)\n",
+    "qc.h(1)\n",
+    "qc.h(0)\n",
+    "\n",
+    "qc.draw(output='mpl')"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<!-- ::: q-block.exercise -->\n",
+    "\n",
+    "### Problem 2\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"enigma-005-quiz-2\") -->\n",
+    "\n",
+    "<!-- ::: .question -->\n",
+    "\n",
+    "Which statement is true if you simplify the circuit for the following situation representing when Constantin does not repaint the statues.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option(correct) -->\n",
+    "\n",
+    "1. No gate is left on $q_0$ and its state is 0.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "2. No gate is left on $q_1$ and its state is 1.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "3. Nothing changes.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "4. Only one H gate is left on $q_0$ and it is in a state superposition.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem 3 + 4\n",
+    "\n",
+    "Write a circuit that would allow you to randomly determine which oracle is used i.e. which vandal used which scenario.\n",
+    "\n",
+    "Use the following legend:\n",
+    "\n",
+    "1. Qubit 0 is the park: O is Ozis and 1 is Ida\n",
+    "2. Qubit 1 is the colour: 0 is orange and 1 in indigo\n",
+    "3. Qubit 2 determines which vandal is involved: 0 is Constantin, 1 is Equilibria\n",
+    "4. Qubit 3 determines which scenario is present: 0 means no painting for Constantin and Ozis Park statue is painted for Equilibria, 1 means both parks are painted for Constatin and Ida Park statue is painted for Equilibria.\n",
+    "\n",
+    "Note: The MCX gate is the multi-control X gate:\n",
+    "\n",
+    "```python\n",
+    "problem_qc.append(gate, [0, 1, 2, 3])\n",
+    "```\n",
+    "\n",
+    "where $q_0$, $q_1$, $q_2$ are the control qubits and $q_3$ is the target qubit."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "goals": [
+     {
+      "id": "enigma-005-quiz-3"
+     }
+    ],
+    "grader_answer": "problem_qc",
+    "grader_id": "quantum_enigmas/enigma_005_problem_3"
+   },
+   "outputs": [],
+   "source": [
+    "from qiskit import QuantumCircuit\n",
+    "from qiskit.circuit.library import MCXGate\n",
+    "\n",
+    "# Start your work here.\n",
+    "# Your circuit MUST be named problem_qc\n",
+    "# Remove all barriers before submitting for grading\n",
+    "\n",
+    "problem_qc = QuantumCircuit(4, 4)\n",
+    "gate = MCXGate(3)\n",
+    " \n",
```

Comment:
Add:
problem_qc.measure(0,0)
problem_qc.measure(2,2)
so that we can run the code in the next box